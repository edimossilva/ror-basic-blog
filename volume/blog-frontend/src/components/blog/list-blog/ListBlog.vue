<template lang="">
  <div>
    <p>List Blogs</p>

    <table>
      <thead>
        <tr>
          <th>id</th>
          <th>name</th>
          <th>private</th>
          <th>show</th>
          <th>Delete</th>

        </tr>
      </thead>
      <tbody>
        <tr v-for="blog in blogs" v-bind:key="blog.id">
          <td>{{blog.id}}</td>
          <td>{{blog.name}}</td>
          <td>{{blog.is_private}}</td>
          <td><button @click="redirectToShowBlog(blog.id)" class="ListBlog__showBlog-button-js">Show</button></td>
          <td><button @click="deleteBlog(blog.id)" class="ListBlog__deleteBlog-button-js">Delete</button></td>
        </tr>
      </tbody>
    </table>
    <button @click="redirectTocreateBlog()">Create Blog</button>
  </div>
</template>

<script>
import blogsApi from "../../../services/blogs_api";

export default {
  data() {
    return {
      blogs: []
    };
  },

  methods: {
    deleteBlog(blogId) {
      const { getBlogs, showErrorMessage } = this;
      const params = { id: blogId };

      blogsApi
        .deleteBlog({ params, token: this.$store.getters.authToken })
        .then(getBlogs)
        .catch(showErrorMessage);
    },
    getBlogs() {
      const { onGetBlogsSuccess, showErrorMessage } = this;

      blogsApi
        .getBlogs({ token: this.$store.getters.authToken })
        .then(onGetBlogsSuccess)
        .catch(showErrorMessage);
    },

    onGetBlogsSuccess({ data: { data: blogs } }) {
      // debugger; // eslint-disable-line
      this.blogs = blogs;
    },

    showErrorMessage(error) {
      debugger; // eslint-disable-line

      console.log(`error ${error}`);
    },

    redirectToShowBlog(blogId) {
      this.$router.push({ name: "showBlog", params: { id: blogId } });
    },

    redirectTocreateBlog() {
      this.$router.push({ name: "createBlog" });
    }
  },

  mounted: function() {
    this.getBlogs();
  }
};
</script>

<style scoped></style>