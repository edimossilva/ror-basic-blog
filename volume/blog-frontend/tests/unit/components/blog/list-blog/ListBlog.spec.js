import { mount } from '@vue/test-utils'
import ListBlog from '@/components/blog/list-blog/ListBlog'

describe('ListBlog.vue', () => {
  it('is a valid component', () => {

    const wrapper = mount(ListBlog, {
      methods: { getBlogs: jest.fn() },
    });

    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('render blogs', () => {
    const mockBlogs = [
      { id: 1, name: "mockName1", is_private: true },
      { id: 2, name: "mockName2", is_private: false },
    ]

    const wrapper = mount(ListBlog, {
      methods: { getBlogs: jest.fn() },
      data() { return { blogs: mockBlogs } }
    });

    expect(wrapper).toMatchSnapshot()
  })
})
