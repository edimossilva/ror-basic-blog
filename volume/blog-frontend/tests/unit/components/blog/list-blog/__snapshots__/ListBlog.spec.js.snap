// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBlog.vue render blogs 1`] = `
<div>
  <p>List Blogs</p>
  <table>
    <thead>
      <tr>
        <th>id</th>
        <th>name</th>
        <th>private</th>
        <th>show</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>1</td>
        <td>mockName1</td>
        <td>true</td> <button class="ListBlog__showBlog-button-js">Show</button>
      </tr>
      <tr>
        <td>2</td>
        <td>mockName2</td>
        <td>false</td> <button class="ListBlog__showBlog-button-js">Show</button>
      </tr>
    </tbody>
  </table> <button class="ListBlog__createBlog-button-js">Create</button>
</div>
`;
